app/assets/javascripts/discourse/routes/unknown.js.es6

```diff
@@ -1,8 +1,28 @@
 import DiscourseRoute from "discourse/routes/discourse";
+import RSVP from "rsvp";
 import { ajax } from "discourse/lib/ajax";
 
 export default DiscourseRoute.extend({
-  model() {
-    return ajax("/404-body", { dataType: "html" });
+  model(params) {
+    const bodyReq = ajax("/404-body", { dataType: "html" });
+    const testReq = ajax("/permalink-check.json", {
+      data: { path: params.path }
+    });
+    const self = this;
```

Comment:
We prefer to use the fat arrow rather than `self` - looks like it should be possible in this case.